### PR TITLE
test: refactor `StringCoder` tests

### PR DESCRIPTION
**Pull request by danielbate**
Related to #753

This PR will:
- Introduce a seperate test file for the service under test
- Seperate test cases for `encode` and `decode`
- Remove snapshot tests to bring expectations into the same test file
- Remove redundant tests to more clearly report on code coverage

### CODE REVIEW

**Comment by danielbate** [comment]
Screenshot 2023-05-02 at 17 12 45

File: packages/abi-coder/src/coders/string.test.ts [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+  //   it('should throw when encoding a string that is too small', () => {
+  //     const coder = new StringCoder(1);
+  //     const invalidInput = STRING_MIN_DECODED;
+
+  //     expect(() => {
+  //       coder.encode(invalidInput);
+  //     }).toThrow();
+  //   });
+});
```
Comment (danielbate) [inline packages/abi-coder/src/coders/string.test.ts:92]: @FuelLabs/sdk-ts should we be throwing for these conditions? Other coders throw for this behaviour, wheras `StringCoder` will instead just slice the input by the length that is past to the construtor.

Comment (arboleya) [reply packages/abi-coder/src/coders/string.test.ts:92]: My instinct says yes — we should throw.

Not sure how the Rust SDK handles this.

> cc @digorithm.

**Review by arboleya — APPROVED** [review]
Approving despite [this](https://github.com/FuelLabs/fuels-ts/pull/942/files#r1182820343). Curious to see what the outcome will be. 🤔

**Review by Dhaiwat10 — APPROVED** [review]
🪄

**Pull request closed without merging**